drivers/usb/device/usb_dc_numaker.c

```diff
@@ -0,0 +1,1901 @@
+	struct usb_dc_numaker_device *dev = (struct usb_dc_numaker_device *)arg1;
+	struct usb_dc_numaker_data *data = dev->data;
+
+	struct m_numaker_usbd_msg msg;
+
+	while (true) {
+		if (k_msgq_get(&data->msgq, &msg, K_FOREVER)) {
+			continue;
+		}
+
+		if (msg.msg_type == NUMAKER_USBD_MSG_TYPE_SW_RECONN) {
+			/* S/W reconnect for error recovery */
+			m_numaker_usbd_lock(dev);
+			m_numaker_usbd_sw_reconnect(dev);
+			m_numaker_usbd_unlock(dev);
+		} else if (msg.msg_type == NUMAKER_USBD_MSG_TYPE_CB_STATE) {
+			/* Interrupt bottom half processing for bus reset */
+			if (msg.cb_state.status_code == USB_DC_RESET) {
+				m_numaker_usbd_lock(dev);
+				m_numaker_usbd_bus_reset_bh(dev);
+				m_numaker_usbd_unlock(dev);
+			}
+
+			/* NOTE: Don't run callback with our mutex locked, or we may encounter
+			 * deadlock because the Zephyr USB device stack can have its own
+			 * synchronization.
+			 */
+			if (data->status_cb) {
+				/* Disable too many SOF message log */
+				if (msg.cb_state.status_code != USB_DC_SOF) {
+					LOG_DBG("Status callback: status_code=%d",
+						msg.cb_state.status_code);
+				}
+				data->status_cb(msg.cb_state.status_code, NULL);
+			} else {
+				LOG_WRN("No status callback: status_code=%d",
+					msg.cb_state.status_code);
+			}
+		} else if (msg.msg_type == NUMAKER_USBD_MSG_TYPE_CB_EP) {
+			uint8_t ep_addr = msg.cb_ep.ep_addr;
+
+			/* Bind EP context to EP address */
+			struct numaker_usbd_ep *ep_cur = numaker_usbd_ep_mgmt_bind_ep(dev, ep_addr);
+
+			if (!ep_cur) {
+				LOG_ERR("Bind EP context: ep_addr=0x%02x", ep_addr);
+				continue;
+			}
+
+			/* Interrupt bottom half processing for EP */
+			m_numaker_usbd_lock(dev);
+			numaker_usbd_ep_bh(ep_cur, msg.cb_ep.status_code);
+			m_numaker_usbd_unlock(dev);
+
+			/* NOTE: Same as above, don't run callback with our mutex locked */
+			if (ep_cur->cb) {
+				LOG_DBG("EP callback: ep_addr=0x%02x, status_code=%d", ep_addr,
+					msg.cb_ep.status_code);
+				ep_cur->cb(ep_addr, msg.cb_ep.status_code);
+			} else {
+				LOG_WRN("No EP callback: ep_addr=0x%02x, status_code=%d", ep_addr,
+					msg.cb_ep.status_code);
+			}
+		}
+	}
+}
+
+/* ISR */
+static void usb_dc_numaker_isr(struct usb_dc_numaker_device *dev)
+{
+	const struct usb_dc_numaker_config *config = dev->config;
+	struct usb_dc_numaker_data *data = dev->data;
+	struct numaker_usbd_ep_mgmt *ep_mgmt = &data->ep_mgmt;
+
+	struct m_numaker_usbd_msg msg = {0};
+
+	uint32_t volatile usbd_intsts = USBD_GET_INT_FLAG();
+	uint32_t volatile usbd_bus_state = USBD_GET_BUS_STATE();
+
+	/* USB plug-in/unplug */
+	if (usbd_intsts & USBD_INTSTS_FLDET) {
+		/* Floating detect */
+		USBD_CLR_INT_FLAG(USBD_INTSTS_FLDET);
+
+		if (USBD_IS_ATTACHED()) {
+			/* USB plug-in */
+			USBD_ENABLE_USB();
+
+			/* Message for bottom-half processing */
+			msg.msg_type = NUMAKER_USBD_MSG_TYPE_CB_STATE;
+			msg.cb_state.status_code = USB_DC_CONNECTED;
+			m_numaker_usbd_send_msg(dev, &msg);
+
+			LOG_INF("USB plug-in");
```

Comment:
you probably want to convert most or all `LOG_INF` to `LOG_DBG`

Reply:
still relevant

Reply:
removed verbose `LOG_DBG`. replaced most `LOG_INF` with `LOG_DBG`